src/renderer/tvgPaint.cpp

```diff
@@ -40,6 +40,47 @@
         default: ret = {}; \
     }
 
+namespace {
+    enum ViewportType: uint8_t
+    {
+        General = 0,
+        AxisAligned,
+        None
+    };
+}
+
+static ViewportType _updateViewport(Paint* const paint, RenderMethod* const renderer, const RenderTransform* const transform, Array<RenderData>& clips, RenderUpdateFlag pFlag, RenderData& rd)
```

Comment:
Viewport is a scene function. It probably could intersect the current viewport with the scene viewport in scene layer?